src/Serilog.Sinks.AzureDataExplorer/Sinks/AzureDataExplorerDurableSink.cs

```diff
@@ -0,0 +1,190 @@
+                        }
+                    }
+                }).ToList();
+            }
+            else
+            {
+                ingestionMapping.IngestionMappings = SDefaultIngestionColumnMapping;
+            }
+
+            var kcsb = options.GetKustoConnectionStringBuilder();
+            m_kustoQueuedIngestClient = KustoIngestFactory.CreateQueuedIngestClient(kcsb);
+            m_sink = new LoggerConfiguration()
+                .MinimumLevel.Verbose()
+                .WriteTo.File(new CompactJsonFormatter(),
+                    options.BufferBaseFileName + FileNameSuffix,
+                    restrictedToMinimumLevel: LevelAlias.Minimum,
+                    fileSizeLimitBytes: options.BufferFileSizeLimitBytes,
+                    levelSwitch: options.BufferFileLoggingLevelSwitch,
+                    flushToDiskInterval: TimeSpan.FromSeconds(10),
+                    rollingInterval: options.BufferFileRollingInterval,
+                    rollOnFileSizeLimit: true,
+                    retainedFileCountLimit: options.BufferFileCountLimit,
+                    encoding: Encoding.UTF8
+                ).CreateLogger();
+
+            var payloadReader = new AzureDataExplorerPayloadReader(
+                rollingInterval: options.BufferFileRollingInterval);
+
+            m_shipper = new LogShipper<List<LogEvent>>(
+                bufferBaseFilename: options.BufferBaseFileName,
+                batchPostingLimit: options.BatchPostingLimit,
+                period: options.BufferLogShippingInterval ?? TimeSpan.FromSeconds(5),
+                eventBodyLimitBytes: options.SingleEventSizePostingLimit,
+                levelControlSwitch: options.BufferFileLoggingLevelSwitch,
+                payloadReader: payloadReader,
+                bufferSizeLimitBytes: options.BufferFileSizeLimitBytes,
+                rollingInterval: options.BufferFileRollingInterval,
+                ingestClient: m_kustoQueuedIngestClient,
+                formatProvider: formatProvider,
+                databaseName: databaseName,
+                tableName: tableName,
+                ingestionMapping: ingestionMapping,
+                flushImmediately: flushImmediately
+            );
+        }
+
+        public void Emit(LogEvent logEvent)
+        {
+            // This is a lagging indicator, but the network bandwidth usage benefits
+            // are worth the ambiguity.
+            if (m_shipper.IsIncluded(logEvent))
+            {
+                m_sink.Write(logEvent);
+            }
+        }
+
+        #region IDisposable methods
+
+        public void Dispose()
+        {
+            Dispose(true);
+            GC.SuppressFinalize(this);
+        }
+
+        private void Dispose(bool disposing)
+        {
+            if (m_disposed)
+            {
+                return;
+            }
+
+            if (disposing)
+            {
+                m_sink.Dispose();
+                m_shipper.Dispose();
+            }
```

Comment:
Set the fields to null?

Reply:
I was setting it null but the E2Etestcases were failing due to this, hence had to remove the null setters.